llvm/lib/CheerpWriter/CheerpWasmWriter.cpp

```diff
@@ -692,7 +711,27 @@ void CheerpWasmWriter::encodePredicate(const llvm::Type* ty, const llvm::CmpInst
 	assert(ty->isIntegerTy() || ty->isPointerTy() || ty->isVectorTy());
 	const Type* elementType = nullptr;
 	if (ty->isVectorTy())
+	{
 		elementType = cast<VectorType>(ty)->getElementType();
+		if (elementType->isIntegerTy(64))
+		{
+			if (predicate == CmpInst::ICMP_EQ)
+				encodeInst(WasmSIMDOpcode::I64x2_EQ, code);
+			else if (predicate == CmpInst::ICMP_NE)
+				encodeInst(WasmSIMDOpcode::I64x2_NE, code);
+			else if (predicate == CmpInst::ICMP_SLT)
+				encodeInst(WasmSIMDOpcode::I64x2_LT_S, code);
+			else if (predicate == CmpInst::ICMP_SGT)
+				encodeInst(WasmSIMDOpcode::I64x2_GT_S, code);
+			else if (predicate == CmpInst::ICMP_SLE)
+				encodeInst(WasmSIMDOpcode::I64x2_LE_S, code);
+			else if (predicate == CmpInst::ICMP_SGE)
+				encodeInst(WasmSIMDOpcode::I64x2_GE_S, code);
+			else
+				llvm::report_fatal_error("Unsupported predicate for 64-bit icmp");
```

Comment:
I don't remember whether the other 4 cases are handled (via lowering?) or whether that has to be somehow implemented here (potentially signed comparsion between op0 ^ 0x80000000 and op1 ^ 0x8000000)

Reply:
I'm not able to produce a test that triggers this case. Made a note for later.